src/domain/user/create-user.use-case.ts

```diff
@@ -1,16 +1,30 @@
 import { AppDataSource } from '../../data-source';
 import { User } from '../../data/db';
+import { emailValidator, passwordValidator } from '../../data/validators';
 import { UserModel } from '../../model';
 
-export async function createUserRow(data: UserModel) {
-  const userRepository = AppDataSource.getRepository(User);
-  const user = new User();
+interface CreateUserResponse {
+  data?: Promise<User>;
+  error?: boolean;
+}
+
+export async function createUserRow(input: UserModel): Promise<CreateUserResponse> {
```

Comment:
ja que estamos seguindo minimamente a arquitetura e padronizacao de codigo, os use-cases usamos um sufixo `UseCase` no final do nome, ou seja:

```suggestion
export async function createUserUseCase(input: UserModel): Promise<CreateUserResponse> {
```